providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStore.java

```diff
@@ -103,6 +107,7 @@ private void streamerListener(final Connector connector) throws InterruptedExcep
                         Map<String, FeatureFlag> flagMap = FlagParser.parseString(take.getData(), throwIfInvalid);
                         writeLock.lock();
                         try {
+                            updateChangedFlags(flagMap);
```

Comment:
I think this produces side effects. Rather than manipulating the `changed flags` within the method, we should return a list and reassign it here. Ultimately, the process creates a fresh list anyway, with clearing up front.

Reply:
will implement this